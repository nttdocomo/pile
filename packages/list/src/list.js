import * as React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import {
  compose,
  prefixClsProperty,
  sizeProperty,
  sizes,
} from '@pile/shared';
import { IfComponent } from '@pile/condition';
import { compose as composed, withState, withHandlers } from 'recompose'

const List = enhanced(({
  prefixCls, className, size, value, onChange,
  ...props
}) => {
  const cls = classNames({
    [`${prefixCls}-radio`]: true,
    [className]: className,
    [`is-${size}`]: size,
  });

  return (
    <div className={cls} onClick={onChange}>
      <label className="label" htmlFor={props.id}>
        <span id={props.id} className={classNames('icon', props.radioType)} />
        {props.label + '12344'}
      </label>
    </div>)
})

export default enhance(List);