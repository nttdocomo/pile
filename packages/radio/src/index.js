import Radio from './radio';

export default Radio
// export { default as Radio } from './radio';
