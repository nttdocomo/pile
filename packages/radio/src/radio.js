import * as React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import {
  compose,
  prefixClsProperty,
  sizeProperty,
  sizes,
} from '@pile/shared';
import { IfComponent } from '@pile/condition';
import { compose as composed, withState, withHandlers } from 'recompose'

const enhanced = composed(
  withState('value', 'updateValue', ''),
  withState('id', 'updateId', ''),
  withHandlers({
    onChange: props => event => {
      props.updateValue(event.target.value)
    }
  })
)


const Radio = enhanced(({
  prefixCls, className, size, value, onChange,
  ...props
}) => {
  const cls = classNames({
    [`${prefixCls}-radio`]: true,
    [className]: className,
    [`is-${size}`]: size,
  });

  return (
    <div className={cls} onClick={onChange}>
      <label className="label" htmlFor={props.id}>
        <span id={props.id} className={classNames('icon', props.radioType)} />
        {props.label + '12344'}
      </label>
    </div>)
})

// Button.propTypes = {
//   children: PropTypes.oneOfType([
//     PropTypes.arrayOf(PropTypes.node),
//     PropTypes.node,
//   ]).isRequired,
//   type: PropTypes.oneOf(['default', 'primary', 'success', 'info', 'warning', 'danger']),
//   nativeType: PropTypes.oneOf(['button', 'submit', 'reset']),
//   block: PropTypes.bool,
//   disabled: PropTypes.bool,
//   line: PropTypes.bool,
//   icon: PropTypes.oneOfType([
//     PropTypes.string,
//     PropTypes.node,
//   ]),
//   loading: PropTypes.bool,
//   text: PropTypes.bool,
//   href: PropTypes.string,
//   radius: PropTypes.bool,
//   circle: PropTypes.bool,
// };

// Button.defaultProps = {
//   type: 'default',
//   nativeType: 'button',
//   block: false,
//   disabled: false,
//   line: false,
//   icon: null,
//   loading: false,
//   text: false,
//   href: null,
//   radius: true,
//   circle: false,
// };

const enhance = compose(
  sizeProperty([sizes.SMALL, sizes.LARGE]),
  prefixClsProperty,
);

export default enhance(Radio);