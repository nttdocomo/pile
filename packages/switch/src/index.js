import classnames from 'classnames'
import React from 'react'
import PropTypes from 'prop-types'
import { compose, prefixClsProperty } from '@pile/shared';

class Switch extends React.Component {
  constructor(props) {
    super(props)
    this.onClick = this.onClick.bind(this)
    this.getMinusedSize = this.getMinusedSize.bind(this)
    const { checked, disabled } = props
    this.state = {
      checked: !!checked,
      disabled: !!disabled,
    }
  }

  onClick (e) {
    let { checked, disabled } = this.state
    if (disabled) {
      return
    }
    checked = !checked
    this.props.onChange(checked)
    this.setState({
      checked
    })
  }

  /**
   * 将 CSS 长度单位减去相应数值并转换
   * 如 getMinusedSize(1) => 1px
   * 如 getMinusedSize(3, 1) => 2px
   * @param {*} size 
   * @param {*} minusNum 
   */
  getMinusedSize (size, minusNum = 0) {
    if (typeof size === 'number') {
      return size - minusNum + 'px'
    }
    if (typeof size !== "string") {
      return JSON.stringify(size)
    }
    const reg = /\d+/
    const transedSize = size.replace(reg, (matchNum) => {
      return Number(matchNum) - minusNum
    })
    // 如果输入了非数字字符串，则直接 return
    if (!/^\d+$/.test(size)) {
      return transedSize
    }
    return transedSize + 'px'
  }

  render() {
    let {
      prefixCls, checkedColor,
      name, color, width, height
    } = this.props
    const { checked, disabled } = this.state

    const transedWidth = this.getMinusedSize(width)
    const transedHeight = this.getMinusedSize(height)
    const minusedHeight = this.getMinusedSize(height, 2)
    const borderRadius = transedHeight

    const divWrapCSS = classnames({
      [`${prefixCls}-switch--wrap`]: true,
      isDisabled: disabled,
    })

    const divWrapStyle = {
      backgroundColor: checked ? checkedColor : color,
      width: transedWidth,
      height: transedHeight,
      borderRadius,
    }

    const divInnerStyle = {
      width: minusedHeight,
      height: minusedHeight,
      borderRadius: minusedHeight,
      border: checked ? `1px solid ${checkedColor}` : `1px solid ${color}`,
      left: !checked ? '1px' : '100%',
      marginLeft: !checked ? 0 : '-' + this.getMinusedSize(height, 1),
    }

    return (
      <div className={divWrapCSS} style={divWrapStyle} onClick={this.onClick}>
        <input type="checkbox" name={name} />
        <div style={divInnerStyle}></div>
      </div>
    )
  }
}

Switch.defaultProps = {
  checked: false,
  disabled: false,
  name: '',
  color: '#ccc',
  checkedColor: '#343c5c',
  width: '50px',
  height: '30px',
  onChange: () => {},
}

Switch.propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  name: PropTypes.string,
  color: PropTypes.string,
  width: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  height: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  onChange: PropTypes.func,
}

const enhance = compose(
  prefixClsProperty,
);

export default enhance(Switch);