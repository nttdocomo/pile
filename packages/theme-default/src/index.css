@import 'normalize.css';
@import 'icon.css';
@import 'button.css';
@import 'alert.css';
@import 'mask.css';
@import 'toast.css';
@import 'switch.css';
@import 'radio.css';
