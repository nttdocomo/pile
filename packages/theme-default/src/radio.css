@import 'theme';
@ns pile {
    @c radio{
        color: blue;
        font-size: 23px;
        @d content{
            color: var(--color-white);
            font-size: var(--font-size-base);
            line-height: 20px;
            display: inline-block;
        }
    }
}