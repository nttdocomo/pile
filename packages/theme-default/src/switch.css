@import 'theme';

@ns pile {
  @c switch {
    @m wrap {
      position: relative;

      /* &.bgColor {
        background-color: #ccc;
      }

      &.wrap-shape {
        width: 50px;
        height: 30px;
        border-radius: 30px;
      } */
      &.isDisabled {
        opacity: 0.4;
      }

      & > input {
        width: 100%;
        height: 100%;
        outline: none;
        -webkit-appearance: none;
        -moz-appearance: none;
        appearance: none;


        /* &.input-shape:before {
          width: 30px;
          height: 30px;
          border-radius: 30px;
        } */

        /* &:before {
          background-color: white;
          width: attr(data-width px);
          height: attr(data-width px);
          border-radius: attr(data-width px);
          content: '';
          position: absolute;
          border: 1px solid #ccc;
          box-sizing: border-box;
          transition: all .3s;
        } */
      }

      & > div {
        width: 30px;
        height: 30px;
        position: absolute;
        background-color: white;
        border-radius: 30px;
        top: 1px;
        box-sizing: border-box;
        transition: all .3s;
      }

    }
  }
}