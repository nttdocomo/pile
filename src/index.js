export { default as Icon } from '../packages/icon';
export { default as Button } from '../packages/button';
export { Mask, Toast } from '../packages/toast';
export { default as Switch } from '../packages/switch';
export { default as Radio } from '../packages/radio';
