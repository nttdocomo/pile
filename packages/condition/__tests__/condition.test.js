/**
 * @author renmaomin@126.com
 */
import React from 'react';
import { shallow } from 'enzyme';
import { IfComponent } from '../src';

describe('Condition suite', () => {
  it('render a Condition', () => {
    const wrapper = shallow(<IfComponent when>{() => <p>hello IfComponent</p>}</IfComponent>);
    expect(wrapper).toMatchSnapshot();
  });
});
