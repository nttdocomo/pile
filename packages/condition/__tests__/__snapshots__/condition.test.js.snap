// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Condition suite render a Condition 1`] = `
<p>
  hello IfComponent
</p>
`;
